src/transport/ws-transport.js

```diff
@@ -22,7 +23,8 @@ WsTransport.prototype.connect = function (initialCallback) {
     initialCallbackCalled = true;
     initialCallback(err);
   };
-  this.webSocketClient = new WebSocketClient(this.address, [], undefined, undefined, { timeout: this.timeout }, this.websocketClientConfig);
+  var origin = isNode ? "localhost" : undefined;
```

Comment:
I don't think you need the isNode check.  If this code is running in a browser any `Origin` header you set will get overwritten by the browser on the way out.

That being said, the `isNode` check does make it a tad more clear.  However, with a comment linking to the Geth bug I think the `isNode` check is overkill on adding clarity.